openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java

```diff
@@ -261,4 +261,32 @@ public void triggerOnDemandApplication(App application) {
       LOG.error("Failed in running job", ex);
     }
   }
+
+  public void stopApplicationRun(App application) {
+    if (application.getFullyQualifiedName() == null) {
+      throw new IllegalArgumentException("Application's fullyQualifiedName is null.");
+    }
+    try {
+      // Interrupt any scheduled job
+      JobDetail jobDetailScheduled =
+          scheduler.getJobDetail(new JobKey(application.getName(), APPS_JOB_GROUP));
+      if (jobDetailScheduled != null) {
```

Comment:
can we add debug logs here for each of the sops?

Reply:
added